### PR TITLE
Chore/improve ts experience on npm

### PR DESCRIPTION
**Pull request by gcornut**
# General summary

- Rework of the TS compiler (babel based) & config
- Exposed TS types on NPM (for @lumx/core, @lumx/icons & @lumx/react)

### CODE REVIEW

File: packages/lumx-core/src/js/utils.ts [hunk 1]
```diff
@@ -17,10 +16,11 @@ import { ESCAPE_KEY_CODE } from './constants';
 /**
  * Enhance isEmpty method to also works with numbers.
  *
- * @param  {any}     value The value to check.
- * @return {boolean} Weither if the input value is empty or != 0.
+ * @param  value The value to check.
+ * @return Whether if the input value is empty or != 0.
```
Comment (pidupuis) [inline packages/lumx-core/src/js/utils.ts]:
```suggestion
 * @return Whether the input value is empty or != 0.
```

File: packages/lumx-react/src/components/autocomplete/__mockData__/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* tslint:disable object-literal-sort-keys */
```
Comment (pidupuis) [inline packages/lumx-react/src/components/autocomplete/__mockData__/index.ts]: We could remove this rule for good :man_shrugging:

Comment (gcornut) [reply packages/lumx-react/src/components/autocomplete/__mockData__/index.ts]: I'd say we can do that in another PR in which we remove all old rules, remove tslint and install latest version of eslint with the new rules from lumapps-web

Comment (Flo5k5) [reply packages/lumx-react/src/components/autocomplete/__mockData__/index.ts]: Btw we could extract rules from lumapps-web and create a separate package we could reuse in every web repo

**Pull request merged**